python/plugins/processing/algs/qgis/RandomPointsOnLines.py

```diff
@@ -0,0 +1,257 @@
+        randSeed = self.parameterAsInt(parameters, self.RANDOM_SEED, context)
+        MaxTriesPerPoint = self.parameterAsInt(parameters,
+                                               self.MAXTRIESPERPOINT,
+                                               context)
+        fields = QgsFields()
+        fields.append(QgsField('id', QVariant.Int, '', 10, 0))
+
+        (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT,
+                                               context, fields,
+                                               QgsWkbTypes.Point,
+                                               source.sourceCrs(),
+                                               QgsFeatureSink.RegeneratePrimaryKey)
+        if sink is None:
+            raise QgsProcessingException(
+                self.invalidSinkError(parameters, self.OUTPUT))
+
+        totNPoints = 0  # The total number of points generated
+        missedPoints = 0  # The number of misses (too close)
+        featureCount = source.featureCount()
+        total = 100.0 / (pointCount * featureCount) if pointCount else 1
+        if randSeed:
+            random.seed(randSeed)
+        else:
+            random.seed()
+
+        index = QgsSpatialIndex()
+        points = dict()
+
+        maxIterations = pointCount * MaxTriesPerPoint
+        # Go through all the features of the layer
+        for f in source.getFeatures():
+            if feedback.isCanceled():
+                break
+            lineGeoms = []  # vector of line elements (>=1 for multi)
+            fGeom = f.geometry()  # get the geometry
+            if fGeom is None:
+                feedback.reportError('Null geometry - skipping!', False)
+                continue
+            if fGeom.isEmpty():
+                feedback.reportError('Empty geometry - skipping!', False)
+                continue
+            totLineLength = fGeom.length()
+            if fGeom.isMultipart():
+                # Explode multi-part geometry
+                for aLine in fGeom.asMultiPolyline():
+                    lineGeoms.append(aLine)
+            else:  # Single-part - just add the line geom
+                lineGeoms.append(fGeom.asPolyline())
+
+            # Generate points on the line geometry / geometries
+            # number of random points generated for this (multi)line
+            nPoints = 0
+            nIterations = 0  # number of attempts for this geometry
+            while nIterations < maxIterations and nPoints < pointCount:
+                # Try to generate a point
+                if feedback.isCanceled():
+                    break
+                # Get the random "position" along the line for this point
+                randomLength = random.random() * totLineLength
+                # To accumulated line length as we move along the geometry:
+                accLength = 0
+                # Go through the parts to find the random point location
+                for l in lineGeoms:
```

Comment:
This could all be greatly simplified -- you could directly call QgsGeometry::interpolate on the original line string feature's geometry with the desired randomLength. Basically you're re-implementing QgsGeometry::interpolate here.

Reply:
I think I did that in an earlier iteration.  Don't remember why I went for this solution. I will look at it.

Reply:
How'd you go with this? I'd like to avoid the duplicate (python heavy) code